plutus-tx/src/PlutusTx/Enum.hs

```diff
@@ -43,9 +47,24 @@ instance Enum Integer where
   fromEnum x = x
 
   {-# INLINABLE enumFromTo #-}
-  enumFromTo x y
-    | x > y = []
-    | otherwise = x : enumFromTo (succ x) y
+  enumFromTo x lim
```

Comment:
I changed `y` to `lim` so that it looks more like what's in `GHC.Enum`.  That changed a lot of golden files too.